files/default/bin/deploy.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+PUBLIC=0
+
+run_help() {
+  echo
+  echo " NAME:"
+  echo "    $0 - deploy target image to OpenStack for further testing."
+  echo
+  echo " USAGE:"
+  echo "    $0 [-p | -f [FILE]] -n [IMG_NAME] -r [PR_NUM]"
+  echo
+  echo " OPTIONS:"
+  echo "    -p          - Publish (publicly) the target image as '\$IMG_NAME', renaming the old"
+  echo "                  '\$IMG_NAME' to '\$IMG_NAME - \$PR_NUM' and making that image private"
+  echo
+  echo " ARGUMENTS:"
+  echo "    FILE        - Fully built image to be uploaded to OpenStack if [-p] is not"
+  echo "                  specified. The image name will be: '\$IMG_NAME - \$PR_NUM'."
+  echo
+  echo "    IMG_NAME    - Prefix of the new image or name of the image to replace if"
+  echo "                  [-p] is specified."
+  echo
+  echo "    PR_NUM      - GitHub Pull Request number to suffix private images with."
+  echo
+  exit 0
+}
+
+while getopts "hpf:n:r:" opt ; do
+  case ${opt} in
+    h)
+      run_help
+      ;;
+    p)
+      PUBLIC=1
+      ;;
+    f)
+      FILE="$OPTARG"
+      ;;
+    n)
+      IMG_NAME="$OPTARG"
+      ;;
+    r)
+      PR_NUM="$OPTARG"
+      ;;
+  esac
+done
+
+[ -z $1 ] && run_help
+[ -z "$IMG_NAME" ] && echo "Error: IMG_NAME not set. Try '$0 -h'" && exit 1
+[ -z "$PR_NUM" ] && echo "Error: PR_NUM not set. Try '$0 -h'" && exit 1
+
+set -xe
+
+if [ $PUBLIC == 0 ]
+  then
+    [ -z $FILE ] && echo "Error: FILE not set. Try '$0 -h'" && exit 1
```

Comment:
I changed this but it is dependent on where the deploy.sh is in comparison to the image directory. I'm acting as if it will be placed in bin or something but I can change it as needed.